### DISCUSSION TITLE
Running compute shaders on headless servers

### DISCUSSION THREAD

**Discussion by Xylord** [godotengine/godot-proposals#8661] [General Discussions]
Hello!
I think there would be a lot of usefulness in the ability to run compute shaders on headless servers, as GPU-enabled VPS become more common. For example, I'd like to use one to run a godot program running a compute shader-based simulation and outputting the result of that simulation to clients.
From I can tell, the issues with doing this are:
-Compute shaders do not work on headless (dedicated server) exports
-Non headless exports cannot normally run on headless servers
-The only way to run a non-headless godot export on a headless server would be to use VirtualGL to render it to a virtual X display, but VirtualGL doesn't support Vulkan which is used in the Forward+ rendering mode, which is required to use compute shader features.
I'm mainly wondering whether there is some simple solution to this I overlooked, this feature is a non-starter due to some technical/core limitation, or it's possible and implementable, but simply a low priority due to its niche usefulness. If it's the latter, I'd welcome any pointers on how I could approach this problem.

**Comment by AThousandShips** [comment]
The whole point of headless is that there's no rendering backend at all AFAIK, this would need a third option of some kind, it'd also require more rendering backends and drivers that we don't support, adding a lot of complexity, I don't think there's any "out of the box" solution to this

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/5790.
